doc/src/EVM-WASM-Smart-Contracts.md

```diff
@@ -0,0 +1,75 @@
+
+The main OP_CODEs are **SLOAD** to load data, **SSTORE** to write a *256bit* integer to storage and **CALL** to call another contract [full list of OP_CODES](https://github.com/crytic/evm-opcodes). It's via these instructions that EVM interact with the chain.
+
+![image info](pictures/eth-1.png)
+
+##### Gas model
+The incentive model of interacting with EVM is gas. It acts like a fee that is calculated for each instruction you are executing. Every instruction has a certain pre-calculated fee amount, and your total gas is the sum of all the instruction you will execute.  
+
+Please check the entire table of [fees](https://blockgeeks.com/wp-content/uploads/2018/03/image2-2.png)
+A gas fee is only due when interacting with EVM. There is no charge for the space on storage you are using, and it is resulting by having a ton of deadcode (or non-used contracts) staying on ethereum chain forever.
+
+##### Evm process
+
+![image info](pictures/eth2.png)
+
+### Substrate Contract-Pallet
+
+##### Intro
+`pallet-contract` is a module in Substrate (the framework to build blockchain on `Polkadot`). Its purpose is a sandbox environment providing WASM interpreter to execute smart contracts in WASM.
+
+![image info](pictures/WASM1.png)
+
+##### WASM interpreter
+
+At the moment contract-pallet use [wasmi](https://github.com/paritytech/wasmi) as interpreter. Wasmi is a pure interpreter-type WASM virtual machine as execution of smart contract needs a high degree of correctness.
+
+##### Storage rent & Gas
+
+In order to incentives the deletion of unused code on chain, pallet-contract implemented a storage-rent principle.  
+
+Data that persist on chain (the contract code + all of its storage) will be charged. So every smart contract will have a rent to pay (small amount at every block). When the smart contract will not have enough fund to pay the rent the smart contract will become a Tombstone (storage will be deleted).  
+
+The gas system of ethereum (price depending on the complexity of the computation) is still present, but it is charged after the call is executed as it is basically a fee on the time of execution (the more time it takes for the node to execute your call the more you will pay). The `pallet-contract` will define the [amount of gas](https://substrate.dev/docs/en/knowledgebase/smart-contracts/contracts-pallet)
+
+##### Contract code & instance are decoupled
+
+Even though pallet-contract use an account model for their contracts alike ethereum does as well, there is still a big difference:  
+
+When you deploy a WASM smart contract on chain it will only create a hash of the WASM code, that's mean this contract will not have an address, nor an associated storage (so no interaction possible). As opposed to ethereum where every contract code deployed on chain will have a unique address (instance), and an associated storage.  
```

Comment:
```suggestion
When you deploy a WASM smart contract, it will only create a hash of the WASM code, meaning that this contract will have no address, nor will it have some associated storage (so it is impossible to interact with this contract). In Ethereum, though, every contract code deployed on chain has a unique address (instance) and its associated storage.  
```